jp/src/edu/hillel/lesson17/part2/PetrolStation.java

```diff
@@ -0,0 +1,36 @@
+package edu.hillel.lesson17.part2;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PetrolStation {
+    private Map<String, Integer> fuelInventory = new ConcurrentHashMap<>();
+
+    public PetrolStation(Map<String, Integer> initialFuelInventory) {
+        this.fuelInventory.putAll(initialFuelInventory);
+    }
+
+    public void doRefuel(Car car) {
+        try {
+            String fuelType = car.getFuelType();
+            int requiredAmount = car.getFuelAmount();
+
+            System.out.println(car.getModel() + " arrived to refuel " + fuelType + ".");
+            int refuelTime = 3000 + (int) (Math.random() * 7000);
+            Thread.sleep(refuelTime);
+
+            fuelInventory.compute(fuelType, (key, value) -> {
+                int currentAmount = value;
+                int remainingAmount = Math.max(currentAmount - requiredAmount, 0);
```

Comment:
You need to notify somehow that required amount can't be processed or throw exception